jobs/build-podman.Jenkinsfile

```diff
@@ -0,0 +1,379 @@
+        // This is mostly so we can prevent re-downloading the RPMS
+        // and repo metadata and over again in a given day for successive
+        // builds.
+        def cacheTTL = "5h"
+        def force = ""
+        if (params.FORCE) {
+            force = '--force'
+            // Also set cacheTTL to 0.1s to allow users an escape hatch
+            // to force no cache layer usage.
+            cacheTTL = "0.1s"
+        }
+
+        withCredentials([file(credentialsId: 'podman-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage('Build Layered Image') {
+                 parallel arches.collectEntries{arch -> [arch, {
+                     pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                         shwrap("""
+                             cosa remote-build-container \
+                                 --arch $arch --cache-ttl ${cacheTTL} \
+                                 --git-ref $commit ${force} \
+                                 --git-url ${params.PODMAN_MACHINE_GIT_URL} \
+                                 --git-sub-dir podman-image-daily \
+                                 --repo ${params.CONTAINER_REGISTRY_STAGING_REPO} \
+                                 --push-to-registry --auth=\$REGISTRY_SECRET
+                         """)
+                     }
+                 }]}
+            }
+            // Initialize the sessions on the remote builders
+            stage("Initialize Remotes") {
+                parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                    if (arch != "x86_64") {
+                        pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                            archinfo[arch]['session'] = pipeutils.makeCosaRemoteSession(
+                                // The cache size for aarch64 in runvm is insufficient,
+                                // even when considering that all artifacts are around
+                                // the same size as those for x86_64
+                                env: ["OSBUILD_EXPORT_FORCE_NO_PRESERVE_OWNER":"true", "RUNVM_CACHE_SIZE": "25G"],
+                                expiration: "60m",
+                                image: params.COREOS_ASSEMBLER_IMAGE,
+                                workdir: WORKSPACE,
+                            )
+                            withEnv(["COREOS_ASSEMBLER_REMOTE_SESSION=${archinfo[arch]['session']}"]) {
+                                 shwrap("""cosa init --force https://github.com/coreos/fedora-coreos-config.git""")
+                            }
+                        }
+                    } else {
+                         shwrap("""cosa init --force https://github.com/coreos/fedora-coreos-config.git""")
+                    }
+                }]}
+            }
+
+            stage("Build Disk Images") {
+                  parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                        if (arch != "x86_64") {
```

Comment:
We shouldn't need this conditional. `withOptionalExistingCosaRemoteSession()` already knows to directly execute the passed body directly if it's x86_64 (well really, if no `session` is defined). So this would just be:

```
parallel ... {
  pipeutils.withOptional...() {
    generate_disk_images(...)
  }
}
```

And similarly for the other stages.